platform/android/export/export.cpp

```diff
@@ -2274,109 +2529,252 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
+
+		if (err != OK) {
+			EditorNode::add_io_error("Could not fix Android Manifest file\n");
+			return err;
+		}
+
+		APKExportData ed;
+		err = export_project_files(p_preset, save_gradle_project_file, &ed, save_gradle_project_so);
 
-			src_apk = src_apk.strip_edges();
+		if (err != OK) {
+			EditorNode::add_io_error("Could not export project files to gradle project\n");
+			return err;
+		}
+
+		OS::get_singleton()->set_environment("ANDROID_HOME", sdk_path); //set and overwrite if required
+
+		String build_command;
+		String package_name = get_package_name(p_preset->get("package/unique_name"));
+
+#ifdef WINDOWS_ENABLED
+		build_command = "gradlew.bat";
+#else
+		build_command = "gradlew";
+#endif
+
+		String build_path = ProjectSettings::get_singleton()->get_resource_path().plus_file("android/build");
+
+		build_command = build_path.plus_file(build_command);
+
+		Vector<PluginConfig> enabled_plugins = get_enabled_plugins(p_preset);
+		String local_plugins_binaries = get_plugins_binaries(BINARY_TYPE_LOCAL, enabled_plugins);
+		String remote_plugins_binaries = get_plugins_binaries(BINARY_TYPE_REMOTE, enabled_plugins);
+		String custom_maven_repos = get_plugins_custom_maven_repos(enabled_plugins);
+		bool clean_build_required = is_clean_build_required(enabled_plugins);
+
+		List<String> cmdline;
+		if (clean_build_required) {
+			cmdline.push_back("clean");
+		}
+		cmdline.push_back("build");
+		cmdline.push_back("-Pexport_package_name=" + package_name); // argument to specify the package name.
+		cmdline.push_back("-Pplugins_local_binaries=" + local_plugins_binaries); // argument to specify the list of plugins local dependencies.
+		cmdline.push_back("-Pplugins_remote_binaries=" + remote_plugins_binaries); // argument to specify the list of plugins remote dependencies.
+		cmdline.push_back("-Pplugins_maven_repos=" + custom_maven_repos); // argument to specify the list of custom maven repos for the plugins dependencies.
+		cmdline.push_back("-p"); // argument to specify the start directory.
+		cmdline.push_back(build_path); // start directory.
+
+		//Executes the build command
+		int result = EditorNode::get_singleton()->execute_and_show_output(TTR("Building Android Project (gradle)"), build_command, cmdline);
+		if (result != 0) {
+			EditorNode::get_singleton()->show_warning(TTR("Building of Android project failed, check output for the error.\nAlternatively visit docs.godotengine.org for Android build documentation."));
+			return ERR_CANT_CREATE;
+		}
+
+		return OK;
+	}
+
+	virtual Error export_project(const Ref<EditorExportPreset> &p_preset, bool p_debug, const String &p_path, int p_flags = 0) {
+
+		ExportNotifier notifier(*this, p_preset, p_debug, p_path, p_flags);
+
+		String src_apk;
+
+		EditorProgress ep("export", "Exporting for Android", 105, true);
+
+		if (bool(p_preset->get("custom_template/use_custom_build"))) { //custom build
+			return gradle_build(p_preset, p_debug, p_path, src_apk, p_flags);
```

Comment:
Does the code below setting `src_apk` need to be moved up above this line, or should src_apk be removed from the parameter list for gradle_build?

Reply:
Looks like src_apk isn't need as a a param to gradle_build.